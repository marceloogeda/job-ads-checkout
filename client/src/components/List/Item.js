// @flow
import React, { type Node } from 'react';
import styled from 'styled-components';

// Theme
import { colors } from './../../theme/palette';

// Styles
const Item = styled.li`
  border-top: ${props => (props.bordered ? `5px solid ${colors.gray}` : '0')};
  display: flex;
  justify-content: space-between;
  padding: 0.75rem 0;
`;

export type ItemProps = {
  bordered?: boolean,
  children: Node
};

export default ({ bordered, children }: ItemProps) => (
  <Item bordered={bordered}>{children}</Item>
);
