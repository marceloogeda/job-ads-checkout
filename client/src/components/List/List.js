// @flow
import React, { type Node, Component } from 'react';
import styled from 'styled-components';

// Components
import Item from './Item';

// Styles
const List = styled.ul`
  margin: 0;
  padding: 0;
`;

export type ListProps = {
  items: any,
  footer: {},
  renderItem: () => Node
};

export default class extends Component<ListProps> {
  static Item = Item;
  static defaultProps = {
    items: []
  };

  render() {
    const { items, footer, renderItem } = this.props;

    return (
      <List>
        {!!renderItem
          ? renderItem()
          : items.map((item: any, index: number) => (
              <Item key={index}>{item}</Item>
            ))}
        {!!footer && footer}
      </List>
    );
  }
}
