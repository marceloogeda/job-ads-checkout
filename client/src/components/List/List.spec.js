import React from 'react';
import TestRenderer from 'react-test-renderer';

import List from './List';

describe('# List', () => {
  it('should render correctly', () => {
    const testRenderer = TestRenderer.create(<List items={[1, 2, 3, 4, 5]} />);

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('should render a empty list', () => {
    const testRenderer = TestRenderer.create(<List />);

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
