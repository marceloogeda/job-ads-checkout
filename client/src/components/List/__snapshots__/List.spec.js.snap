// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# List should render a empty list 1`] = `
<ul
  className="sc-bwzfXH iCGSIH"
/>
`;

exports[`# List should render correctly 1`] = `
<ul
  className="sc-bwzfXH iCGSIH"
>
  <li
    className="sc-bdVaJa hnOvng"
  >
    1
  </li>
  <li
    className="sc-bdVaJa hnOvng"
  >
    2
  </li>
  <li
    className="sc-bdVaJa hnOvng"
  >
    3
  </li>
  <li
    className="sc-bdVaJa hnOvng"
  >
    4
  </li>
  <li
    className="sc-bdVaJa hnOvng"
  >
    5
  </li>
</ul>
`;
